src/components/main/NewItemModal.tsx

```diff
@@ -55,9 +62,14 @@ const NewItemModal: FC<Props> = ({ open, handleClose }) => {
   const [selectedItemType, setSelectedItemType] = useState<NewItemTabType>(
     ItemType.FOLDER,
   );
-  const [initialItem] = useState<Partial<Item<UnknownExtra>>>({});
+  const [initialItem] = useState<ItemRecord>(convertJs({}));
+
+  // todo: find a way to create this type of literal from the enum values instead of like this...
   const [updatedPropertiesPerType, setUpdatedPropertiesPerType] = useState<{
-    [key: string]: Partial<Item<UnknownExtra>>;
+    [ItemType.FOLDER]: { type: 'folder' };
+    [ItemType.APP]: { type: 'app' };
+    [ItemType.LINK]: { type: 'embeddedLink' };
+    [ItemType.DOCUMENT]: { type: 'document' };
```

Comment:
Use `ItemType` instead instead of strings?

Reply:
This is the type for the state. It is better (I think) to specify the type with string literals so the discriminated union is understood better.